packages/wallet-sdk/src/CoinbaseWalletProvider.ts

```diff
@@ -17,22 +17,29 @@ import { Communicator } from ':core/communicator/Communicator';
 import { SignerType } from ':core/message';
 import { determineMethodCategory } from ':core/provider/method';
 import { hexStringFromNumber } from ':core/type/util';
-import { ScopedLocalStorage } from ':util/ScopedLocalStorage';
+import type { BaseStorage } from ':util/BaseStorage';
+import { ScopedStorage } from ':util/ScopedStorage';
 
 export class CoinbaseWalletProvider extends EventEmitter implements ProviderInterface {
   private readonly metadata: AppMetadata;
   private readonly preference: Preference;
   private readonly communicator: Communicator;
+  private readonly baseStorage?: BaseStorage;
 
   private signer: Signer | null;
 
-  constructor({ metadata, preference: { keysUrl, ...preference } }: Readonly<ConstructorOptions>) {
+  constructor({
+    baseStorage,
```

Comment:
so if i understand this correctly,
the branching, e.g. `const baseStorage = isMobile? mobileStorage : webStorage` would happen outside, which means the consumer of this class will be responsible for it, right?
@vishnumad

Reply:
@bangtoven, going to merge this to unblock mobile sdk progress. Let's revisit encapsulating this logic in the sdk as a followup.